# CS216 Project Proposal: Group 26
#### Jack Lichtenstein, Shannon Houser, Libba Lawrence, and Linda Tang

## Part 1: Introduction and Research Questions

Introducing your topic in general and then defining one or more research questions

Spotify Data ML

Potential questions:
How to Predict Popularity of spotify songs based off of other factors
https://towardsdatascience.com/predicting-popularity-on-spotify-when-data-needs-culture-more-than-culture-needs-data-2ed3661f75f1 

## Part 2: Data Sources

Your proposal should discuss the data you will use to answer your research questions. Be as specific as possible: name the datasets you will use and how you will access them or specify where you will look for the relevant datasets and why you expect to be successful in finding them. You should also briefly justify why the data you plan to obtain will be relevant and appropriate for addressing your research questions. 

The data used for this project comes from an episode of the [Sliced](https://www.notion.so/SLICED-Show-c7bd26356e3a42279e2dfbafb0480073) data science challenge held in summer 2021. The data is accessible publicly on [Kaggle](https://www.kaggle.com/c/sliced-s01e08-KJSEks/data). The competition provides three datasets, each of which we will make use of to answer our research questions:

- train.csv - the training set
- test.csv - the test set
- artists.csv - data about artists in the dataset

The train and test datasets are formatted the same way. We read in `train.csv` and `artists.csv` below. Additionally, we provide an in-depth data description as well.

In [1]:
# import some libraries
import os
import pandas as pd
import numpy as np

In [2]:
# change to one directory back (only run this once)
os.chdir(os.path.normpath(os.getcwd() + os.sep + os.pardir))

In [3]:
# read in spotify data
spotify_train = pd.read_csv("data/train.csv")
spotify_train.head()

id                                    name  popularity  duration_ms  \
0    269                       blun7 a swishland          63     167760.0   
1  27504                Que Me Perdone Tu Señora          42     150640.0   
2  16082                          愛唄~since 2007~          42     242373.0   
3  14585            Let me be your uncle tonight          12     202989.0   
4  14740  Never Going Back Again - 2004 Remaster          40     134400.0   

               artists                  id_artists  danceability  energy  key  \
0      ['tha Supreme']  ['19i93sA0D7yS9dYoVNBqAA']         0.692   0.792    7   
1  ['Manoella Torres']  ['4JRKcLbpjobmoOVoOXPd6y']         0.608   0.447    6   
2        ['whiteeeen']  ['6v3VFX2qIWthj4Lr5QIxts']         0.572   0.782    8   
3         ['Tvíhöfði']  ['6rmrk3Jk0Ecf8fjioCCZmV']         0.855   0.470    7   
4    ['Fleetwood Mac']  ['08GQAI4eElDnROBrJRGE0X']         0.654   0.336    6   

   loudness  speechiness  acousticness  instrumentalness  liveness  valence  \
0    -5.984       0.2450        0.1300          0.000000    0.2550    0.566   
1   -12.151       0.0663        0.3350          0.000035    0.1690    0.673   
2    -5.275       0.0361        0.3060          0.000000    0.3740    0.450   
3    -9.252       0.1620        0.0111          0.000002    0.2640    0.864   
4   -12.825       0.0379        0.8580          0.385000    0.0786    0.947   

     tempo  release_year  release_month  release_day  
0  129.883          2019           11.0         15.0  
1   90.777          2007            3.0         21.0  
2  170.050          2016            3.0          9.0  
3  102.008          2001            1.0          1.0  
4  176.083          1977            2.0          4.0

In [5]:
# read in spotify artist data
spotify_artists = pd.read_csv("data/artists.csv")
spotify_artists.head()

17718

### train.csv/test.csv
- id (Unique identifier of track)
- name (Name of the song)
- popularity (Ranges from 0 to 100)
- duration_ms (Integer typically ranging from 200k to 300k)
- artists (List of artists mentioned)
- id_artists (Ids of mentioned artists)
- danceability (Ranges from 0 to 1)
- energy (Ranges from 0 to 1)
- key (All keys on octave encoded as values ranging from 0 to 11, starting on C as 0, C# as 1 and so on...)
- loudness (Float typically ranging from -60 to 0)
- speechiness (Ranges from 0 to 1)
- acousticness (Ranges from 0 to 1)
- instrumentalness (Ranges from 0 to 1)
- liveness (Ranges from 0 to 1)
- valence (Ranges from 0 to 1)
- tempo (Float typically ranging from 50 to 150)
- release_year (Year of release)
- release_month (Month of year released)
- release_day (Day of month released)

### artists.csv
- id (Id of artist)
- followers (Total number of followers of artist)
- genres (Genres associated with this artist)
- name (Name of artist)
- popularity (Popularity of given artist based on all his/her tracks)

These data are extremely detailed and rich. The `train` set has 21,000 rows, while the `artists` set has 17,718 rows. Note the ease in which we can join the two together as well, specifically using `id_artists` = `id` to join any of `test/train` with `artists`. No doubt, we will be able to answer many interesting questions about how and why a given song is popular/unpopular. We may also be able to explore the similarity of songs within artists and across artists based on a number of the features present in their songs (energy, key, loudness, etc.).

## Part 3: Collaboration Plan

How will you divide responsibilities? 

About how much time do you expect every group member to spend on the project each week, on average?

When and how will you meet? 

What platform(s) will you use to communicate between meetings? 

Where will you store data, code, writing, etc., so that all group members have easy access to shared materials?*
